src/context.rs

```diff
@@ -103,74 +103,62 @@ pub fn get_info() -> BTreeMap<&'static str, String> {
     res
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum ContextError {
```

Comment:
Can having a public error type and matching on errors be avoided?

E.g. return `anyhow::Result<Option<Self>>` from `Context::new`, with `None` meaning the key is probably wrong. Then in `load_accounts` you can use `Context::new(...).with_context(|| "failed to create context from file {}", ..)?`.

If you don't like `Option` you can make your own `enum` type, but errors are something that is either passed upwards with `?` or written to logs if there is a fallback to be tried.

Reply:
Hum, going to have to sleep over this.  I think with `#[error(transparent)]` this is a fairly nice option here that allows matching on just what we care about while still letting you propagate errors.  I'm not sure if nested types are that much nicer.